In [42]:
# Import Dependencies 
import pandas as pd 
import numpy as np 
import matplotlib.pyplot as plt


In [3]:
# Read in csv file and create DataFrame 
vine_df = pd.read_csv("./Resources/vine_table.csv")
vine_df.head()

RTIS3L2M1F5SM  5  0  0.1  N  Y
0   R1ZV7R40OLHKD  5  0    0  N  Y
1  R3BH071QLH8QMC  1  0    1  N  Y
2  R127K9NTSXA2YH  3  0    0  N  Y
3  R32ZWUXDJPW27Q  4  0    0  N  Y
4  R3AQQ4YUKJWBA6  1  0    0  N  Y

In [5]:
# Rename columns because csv was not exported with column names from pgAdmin
vine_df.rename(columns = {'RTIS3L2M1F5SM':'review_id',
                          '5':'star_rating',
                          '0':'helpful_votes',
                          '0.1':'total_votes',
                          'N':'vine',
                          'Y':'verified_purchase'}, inplace=True)
vine_df.head()

review_id  star_rating  helpful_votes  total_votes vine  \
0   R1ZV7R40OLHKD            5              0            0    N   
1  R3BH071QLH8QMC            1              0            1    N   
2  R127K9NTSXA2YH            3              0            0    N   
3  R32ZWUXDJPW27Q            4              0            0    N   
4  R3AQQ4YUKJWBA6            1              0            0    N   

  verified_purchase  
0                 Y  
1                 Y  
2                 Y  
3                 Y  
4                 Y

In [6]:
vine_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1785996 entries, 0 to 1785995
Data columns (total 6 columns):
 #   Column             Dtype 
---  ------             ----- 
 0   review_id          object
 1   star_rating        int64 
 2   helpful_votes      int64 
 3   total_votes        int64 
 4   vine               object
 5   verified_purchase  object
dtypes: int64(3), object(3)
memory usage: 81.8+ MB


## Filter by Total Votes 

* Filter the data and create a new DataFrame or table to retrieve all the rows where the total_votes count is equal to or greater than 20
* do this to pick reviews that are more likely to be helpful and to avoid having division by zero errors later on in code 

In [10]:
total_votes_df = vine_df[vine_df['total_votes'] >= 20]
total_votes_df

review_id  star_rating  helpful_votes  total_votes vine  \
54        R4PKAZRQJJX14            1             21           34    N   
73       R2CI0Y288CC7E2            1             21           35    N   
208      R127WEQY2FM1T3            1            147          175    N   
288      R3EZ0EPYLDA34S            1             14           31    N   
482      R2FJ94555FZH32            2             55           60    N   
...                 ...          ...            ...          ...  ...   
1785687  R3GYUWLD9FWCPS            5             30           30    N   
1785713  R26KS4Q9G04FIV            2              8           21    N   
1785787   RU0J1ZMBCLD27            5             27           29    N   
1785805  R1D69XH2THYKTG            5             18           20    N   
1785934  R3U5C85OV6BUKM            4             38           43    N   

        verified_purchase  
54                      N  
73                      Y  
208                     Y  
288                     Y  
482                     N  
...                   ...  
1785687                 N  
1785713                 N  
1785787                 N  
1785805                 N  
1785934                 N  

[65379 rows x 6 columns]

## Filter Helpful Review Votes Greater Than 50% Over Total Review Votes

* Filter new DataFrame to retrieve all rows where the number of <b> helpful_votes </b> divided by <b> total_votes</b> is equal to or greater than 50%

In [11]:
votes_greater50_percent = total_votes_df.loc[((total_votes_df["helpful_votes"]) / (total_votes_df["total_votes"])*100) >= 50]
votes_greater50_percent

review_id  star_rating  helpful_votes  total_votes vine  \
54        R4PKAZRQJJX14            1             21           34    N   
73       R2CI0Y288CC7E2            1             21           35    N   
208      R127WEQY2FM1T3            1            147          175    N   
482      R2FJ94555FZH32            2             55           60    N   
536      R1U3AR67RE273L            1             51           65    N   
...                 ...          ...            ...          ...  ...   
1785640   RCMDCDJR16IKW            4             27           30    N   
1785687  R3GYUWLD9FWCPS            5             30           30    N   
1785787   RU0J1ZMBCLD27            5             27           29    N   
1785805  R1D69XH2THYKTG            5             18           20    N   
1785934  R3U5C85OV6BUKM            4             38           43    N   

        verified_purchase  
54                      N  
73                      Y  
208                     Y  
482                     N  
536                     Y  
...                   ...  
1785640                 N  
1785687                 N  
1785787                 N  
1785805                 N  
1785934                 N  

[40565 rows x 6 columns]

* (below) Alternative method to achieve same result using pandas 

In [12]:
votes = total_votes_df.loc[total_votes_df.helpful_votes / total_votes_df.total_votes >= 0.5]
votes

review_id  star_rating  helpful_votes  total_votes vine  \
54        R4PKAZRQJJX14            1             21           34    N   
73       R2CI0Y288CC7E2            1             21           35    N   
208      R127WEQY2FM1T3            1            147          175    N   
482      R2FJ94555FZH32            2             55           60    N   
536      R1U3AR67RE273L            1             51           65    N   
...                 ...          ...            ...          ...  ...   
1785640   RCMDCDJR16IKW            4             27           30    N   
1785687  R3GYUWLD9FWCPS            5             30           30    N   
1785787   RU0J1ZMBCLD27            5             27           29    N   
1785805  R1D69XH2THYKTG            5             18           20    N   
1785934  R3U5C85OV6BUKM            4             38           43    N   

        verified_purchase  
54                      N  
73                      Y  
208                     Y  
482                     N  
536                     Y  
...                   ...  
1785640                 N  
1785687                 N  
1785787                 N  
1785805                 N  
1785934                 N  

[40565 rows x 6 columns]

## Filter Reviews Written as Part of Vine Program (Paid) 

* Filter the DataFrame to retrieve all the rows where a review was written as part of the Vine program (paid), vine == 'Y'.


In [13]:
vine_reviews_paid = votes.loc[votes["vine"] == 'Y']
vine_reviews_paid

review_id  star_rating  helpful_votes  total_votes vine  \
32610    R3KKUSGFZWSUIY            5             56           63    Y   
33111    R10FO5UKKVZBK2            3             23           23    Y   
69679     RM4KSGEOR7MU1            5             19           24    Y   
155360    RG7VRMYLEXD23            4             22           26    Y   
239326   R11O4YSCPSNL6L            3             20           26    Y   
...                 ...          ...            ...          ...  ...   
1456861   RLPTVGLU0JQIP            3             42           45    Y   
1463332  R3ASJ9SENYYYI0            5             40           46    Y   
1481161   RNU8PK609WT6P            4            347          362    Y   
1506353   R8YT75NJW0CM9            4             37           40    Y   
1506641  R2EDL85YVF0PRA            5             23           32    Y   

        verified_purchase  
32610                   N  
33111                   N  
69679                   N  
155360                  N  
239326                  N  
...                   ...  
1456861                 N  
1463332                 N  
1481161                 N  
1506353                 N  
1506641                 N  

[94 rows x 6 columns]

## Filter Reviews Written as Part of Vine Program(Unpaid)

* Filter the DataFrame to retrieve all the rows where the review was not part of the Vine program (unpaid), vine == 'N'.

In [14]:
vine_reviews_unpaid = votes.loc[votes["vine"] == "N"]
vine_reviews_unpaid

review_id  star_rating  helpful_votes  total_votes vine  \
54        R4PKAZRQJJX14            1             21           34    N   
73       R2CI0Y288CC7E2            1             21           35    N   
208      R127WEQY2FM1T3            1            147          175    N   
482      R2FJ94555FZH32            2             55           60    N   
536      R1U3AR67RE273L            1             51           65    N   
...                 ...          ...            ...          ...  ...   
1785640   RCMDCDJR16IKW            4             27           30    N   
1785687  R3GYUWLD9FWCPS            5             30           30    N   
1785787   RU0J1ZMBCLD27            5             27           29    N   
1785805  R1D69XH2THYKTG            5             18           20    N   
1785934  R3U5C85OV6BUKM            4             38           43    N   

        verified_purchase  
54                      N  
73                      Y  
208                     Y  
482                     N  
536                     Y  
...                   ...  
1785640                 N  
1785687                 N  
1785787                 N  
1785805                 N  
1785934                 N  

[40471 rows x 6 columns]

## Total Number of Reviews 

In [15]:
# Count of total reviews
votes.count()

review_id            40565
star_rating          40565
helpful_votes        40565
total_votes          40565
vine                 40565
verified_purchase    40565
dtype: int64

In [48]:
# Count of total paid vine reviews
paid_review = len(vine_reviews_paid)
paid_review

94

In [49]:
# Count of total unpaid vine reviews 
unpaid_review = len(vine_reviews_unpaid)
unpaid_review

40471

## Total Number of 5-Star Reviews 

In [24]:
five_star_reviews = votes.loc[votes["star_rating"] == 5]
five_star_reviews


review_id  star_rating  helpful_votes  total_votes vine  \
1401     R36O341WWXXKNP            5             28           31    N   
2506     R29BOS5HMAY1LO            5             88          110    N   
3931      RDX2ZZ46AM343            5             32           34    N   
4132      R60GI4Z1CNGGV            5             17           20    N   
5277     R2FAARI3JQO9XQ            5             29           30    N   
...                 ...          ...            ...          ...  ...   
1785557  R2KWI1FUZLSYKA            5             50           59    N   
1785613  R31UKJUAJNX7XX            5             35           50    N   
1785687  R3GYUWLD9FWCPS            5             30           30    N   
1785787   RU0J1ZMBCLD27            5             27           29    N   
1785805  R1D69XH2THYKTG            5             18           20    N   

        verified_purchase  
1401                    N  
2506                    N  
3931                    Y  
4132                    Y  
5277                    Y  
...                   ...  
1785557                 N  
1785613                 N  
1785687                 N  
1785787                 N  
1785805                 N  

[15711 rows x 6 columns]

In [45]:
five_star_total = len(five_star_reviews)
five_star_total

15711

## Comparing the Vine(Paid) and Non-Vine(Unpaid) Program

### Paid 5 Star Review Count 

In [33]:
paid_5star = vine_reviews_paid.loc[vine_reviews_paid["star_rating"] == 5].count()
paid_5star

review_id            48
star_rating          48
helpful_votes        48
total_votes          48
vine                 48
verified_purchase    48
dtype: int64

In [36]:
paid = len(vine_reviews_paid[vine_reviews_paid['star_rating']==5])
paid

48

### Unpaid 5 Star Review Count

In [37]:
unpaid_5star = vine_reviews_unpaid.loc[vine_reviews_unpaid["star_rating"] == 5].count()
unpaid_5star

review_id            15663
star_rating          15663
helpful_votes        15663
total_votes          15663
vine                 15663
verified_purchase    15663
dtype: int64

In [38]:
unpaid = len(vine_reviews_unpaid[vine_reviews_unpaid['star_rating']==5])
unpaid

15663

<b> Result </b> : There were <b>48</b> paid 5-star reviews and <b>15663</b> unpaid 5-star reviews 

## Percentage of 5-Star Reviews 

### Paid Percentage 

Percentage of 5-star Vine(paid) review: 

In [50]:
paid/paid_review

0.5106382978723404

Percentage of 5-star Non-Vine(unpaid) review: 

In [51]:
unpaid/unpaid_review

0.38701786464381904

<b> Result: </b> Out of all 5-star reviews, <b> 51% were paid </b> for their 5-star reviews, while <b> 38.7% were unpaid </b> for their 5-star reviews. 

## Additional Analysis 

### Average Star Rating for Vine(paid) and Non-Vine(unpaid) Reviews 

1. Average Star Rating for Vine(paid) Reviews: 

In [54]:
paid_star_avg = vine_reviews_paid["star_rating"].mean()
paid_star_avg

4.202127659574468

2. Average Star Rating for Vine(unpaid) Reviews:

In [55]:
unpaid_star_avg = vine_reviews_unpaid["star_rating"].mean()
unpaid_star_avg

3.34765634651973

<b> Result: </b> The average star rating for <b> Paid Vine Reviews = 4.20 </b> and the average star rating for <b> Unpaid Vine Reviews = 3.35. </b>